lib/matplotlib/tests/test_path.py

```diff
@@ -541,3 +541,18 @@ def test_cleanup_closepoly():
         cleaned = p.cleaned(remove_nans=True)
         assert len(cleaned) == 1
         assert cleaned.codes[0] == Path.STOP
+
+
+def test_simplify_closepoly():
+    # The values of the vertices in a CLOSEPOLY should always be ignored,
+    # in favor of the most recent MOVETO's vertex values
+    path_ref = Path([(0, 0), (1, 0), (1, 1), (40, 50)],
```

Comment:
A few quick suggestions for enhancing this test:
* Use a different starting location from (0, 0) since that is inserted with the STOP command.
* It would be good to assert against the expected path directly rather than both of these using `simplify=True`
* Should we also see how `simplify=True` interacts with `remove_nans=True`? (see test above this)
* Is there any issue if the first MOVETO is `(nan, nan)` and then we insert it here?
* Could you test this with a compound path with two CLOSEPOLYs? (making sure the correct MOVETO gets inserted for each subpath)

Reply:
> Should we also see how `simplify=True` interacts with `remove_nans=True`? (see test above this)

I am not sure what you meant by this. It would be helpful if you could elaborate a bit more.

> Is there any issue if the first MOVETO is `(nan, nan)` and then we insert it here?

Having the first MOVETO as `(nan, nan)` causes it to never initialize the initial vertex, and it ends up closing the polygon to (0, 0) which is wrong.

I think that in order to fix this, it could have a `m_has_init` so that when a CLOSEPOLY is encountered, it will be able to verify that the path has a finite initial point.

```
@@ -773,11 +780,11 @@ class PathSimplifier : protected EmbeddedQueue<9>
             }
             m_after_moveto = false;
 
-            if(agg::is_close(cmd)) {
+            if(agg::is_close(cmd) && m_has_init) {
                 /* If we have to close the polygon, replace
                    the vertex with the starting vertex */
-                *x = m_last_startx;
-                *y = m_last_starty;
+                *x = m_initX;
+                *y = m_initY;
             }
```

Refactoring `m_last_startx` to `m_initX` might be better, as that seems to be the variable name in other path converters (such as `NanRemover`)

Reply:
>> Should we also see how simplify=True interacts with remove_nans=True? (see test above this)

> I am not sure what you meant by this. It would be helpful if you could elaborate a bit more.

I wanted to make sure that we weren't doing too much simplification if you pass `simplify=True` to a test similar to the one above this with only nan's. Making sure you still get back some valid path, which I think your other comment with assuring there is at least a single MOVETO does. So nothing to do here now.